ci: deploy a new environment on every pull request

lorenzo-cavazzi opened the pull request:
* Deploy the environment on new pull requests.
* Update the environment on PR changes.
* Cleanup the environment when closing the PR.

re #1165

ciyer commented on client/src/App.js:
```diff
@@ -71,6 +71,7 @@ class App extends Component {
         } />
         <main role="main" className="container-fluid">
           <div key="gap">&nbsp;</div>
+          <h1>TEST CI ACTIONS</h1>
```
Should this code get merged?

lorenzo-cavazzi replied:
Right, I've just removed it. Since we changed the telepresence file, this is not needed even for the testing phase.

ciyer commented:
Just added a commit that makes it possible to run telepresence against a PR environment. E.g., I can do:
```fish
env PR=1166 ./run-telepresence.sh
```
(Might be different in bash)

lorenzo-cavazzi commented:
Great idea!
For bash, the `env` prefix is not necessary.
```
PR=1166 ./run-telepresence.sh
```

Maybe we should refresh the telepresence info in the README file

ciyer commented:
> Maybe we should refresh the telepresence info in the README file

Yes, we should! Could you do that? We can remove the obsolete "make dev" and replace it with a short description of the `run-telepresence` command.

lorenzo-cavazzi commented:
Added readme changes.
I'd like to integrate automatic messages to provide information about
* The CI deployment link
* The current version deployed

Otherwise, it's tedious to get the information from the tests, and it's error-prone since it's not immediately clear whether the latest changes are in the deployed version.
This may be useful https://github.com/marketplace/actions/create-or-update-comment

I converted the PR to a draft since it's a work in progress.

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+    - helm-chart/**
+
+jobs:
+  message-pr-pre-deploy:
```
can you put this inside the deploy job so it doesn't display as an independent check?

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+        token: ${{ secrets.RENKUBOT_GITHUB_TOKEN }}
+        issue-number: ${{ github.event.pull_request.number }}
+        body: |
+          A deplyoment for this PR is being prepared based on the commit ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```
```suggestion
          A deployment for this PR is being prepared based on the commit ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+        comment-id: ${{ steps.findcomment.outputs.comment-id }}
+        edit-mode: replace
+        body: |
+          A deplyoment for this PR is being prepared based on the commit ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```
```suggestion
          A deployment for this PR is being prepared based on the commit ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+        token: ${{ secrets.RENKUBOT_GITHUB_TOKEN }}
+        issue-number: ${{ github.event.pull_request.number }}
+        body: |
+          A deplyoment is ready for this PR :tada:. Currently, the source commit is ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```
```suggestion
          A deployment is ready for this PR :tada:. Currently, the source commit is ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+        renku: "@development"
+        renku_ui: "@${{ github.head_ref }}"
+
+  message-pr-post-deploy:
```
same as above - this can just be a step after deploy, doesn't need to be a separate job

rokroskar commented on .github/workflows/acceptance-tests.yml:
```diff
@@ -0,0 +1,127 @@
+        comment-id: ${{ steps.findcomment.outputs.comment-id }}
+        edit-mode: replace
+        body: |
+          A deplyoment is ready for this PR :tada:. Currently, the source commit is ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```
```suggestion
          A deployment is ready for this PR :tada:. Currently, the source commit is ${{ steps.latestcommit.outputs.latest_commit_sha }}.
```

rokroskar reviewed (changes requested):
🚀

lorenzo-cavazzi commented:
Thank you for the reviews @rokroskar and @ciyer
I added a few bits to notify about failures and teardown.

If the code is fine, I will try to close this PR to test the last case, and push this to a new PR (with squashed commits) to test the whole thing again from scratch.

rokroskar commented:
Sounds good to me @lorenzo-cavazzi! Just please don't forget to fix the typos :)

lorenzo-cavazzi commented:
I don't know how those sneaked in again. I should stop using `git rebase` :facepalm: 
Fixed (hopefully)

ciyer reviewed (approved):
Can't wait to start using this!

vfried commented:
Just had a look at these changes, it looks great!!! 

As Sekhar said, can't wait to have this in master :)

lorenzo-cavazzi commented:
Testing close action

lorenzo-cavazzi commented:
Ok this looks good.

![Screenshot_20201214_142527](https://user-images.githubusercontent.com/43481553/102086331-38eb1600-3e18-11eb-8d3b-9ae67f1ba2d0.png)

I'll rebase the commits and open a new PR to quickly test again the whole thing in a brand new PR.
